spec/interators/users/upload_avatar/upload_file_spec.rb

```diff
@@ -0,0 +1,43 @@
+RSpec.describe Users::UploadAvatar::UploadFile do
+  subject(:call) { described_class.call(user: user, avatar: avatar) }
+
+  let(:user) { create :user }
+  let(:valid_avatar) { fixture_file_upload("avatar.png", "image/png") }
+  let(:invalid_avatar) { fixture_file_upload("example.pptx") }
+
+  describe ".call" do
+    context "when a valid avatar is provided" do
+      let(:avatar) { valid_avatar }
+      it "attaches the avatar to the user" do
+        expect { call }.to change { user.reload.avatar.attached? }.from(false).to(true)
+      end
+    end
+
+    context "when an invalid avatar is provided" do
+      let(:avatar) { invalid_avatar }
+
+      it "does not attach the avatar" do
+        expect { call }.not_to change { user.reload.avatar.attached? }
+      end
+
+      it "fails with an error message" do
+        result = call
+        expect(result).to be_a_failure
+        expect(result.errors).to eq(["Avatar has an invalid content type"])
+      end
+    end
+
+    context "when no avatar is provided" do
+      let(:avatar) { nil }
+
+      it "does not attach an avatar" do
+        expect { call }.not_to change { user.reload.avatar.attached? }
+      end
+
+      it "succeeds without errors" do
+        result = call
+        expect(result).to be_a_success
```

Comment:
Не супер важно но так можно было
```suggestion
        expect(call).to be_a_success
```